src/v/security/krb5.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include <krb5/krb5.h>
+
+#include <string_view>
+
+namespace security::krb5 {
+
+class krb5_context_view {
+public:
+    explicit krb5_context_view(const ::krb5_context* ctx)
+      : _ctx(ctx) {}
+
+    ::krb5_context operator*() const { return *_ctx; }
+    explicit operator const ::krb5_context*() { return _ctx; }
+
+private:
+    const ::krb5_context* _ctx;
+};
+
+class krb5_context {
+public:
+    krb5_context() = default;
+    krb5_context(krb5_context&&) = delete;
+    krb5_context(const krb5_context&) = delete;
+    krb5_context operator=(krb5_context&&) = delete;
+    krb5_context operator=(const krb5_context&) = delete;
+
+    ~krb5_context() noexcept { reset(); }
+
+    void reset() noexcept {
+        if (_ctx != nullptr) {
+            ::krb5_free_context(_ctx);
```

Comment:
should we run all this on the alien thread? the code paths are complicated, so it's not clear it is ever called, but it didn't take long tracing through the source code for this function to find a call to `asprintf` and `fopen`, etc...

Reply:
:alien: thread for any use case at the moment